plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java

```diff
@@ -152,20 +152,9 @@
   }
 
   /**
-   * Waits until no error markers are found in the workspace. Returns true unless it times out
-   * after 300 seconds. Returns false if it times out.
-   * 
-   * This method does not wait for pending build and validation jobs, so it is usually required to
-   * call methods such as {@link #waitForProjects} beforehand.
-   */
-  public static boolean waitUntilNoBuildErrors() throws CoreException {
-    IProject[] projects = getWorkspace().getRoot().getProjects();
-    return waitUntilNoBuildErrors(projects);
-  }
-
-  /**
-   * Waits until no error markers are found for the projects. Returns true unless it times out after
-   * 300 seconds. Returns false if it times out.
+   * Waits until no error markers are found for the specified projects or any projects in the
+   * workspace if no project is specified. Returns true unless it times out after 300 seconds.
```

Comment:
I think it may sound better by separating out the if-no-projects-specified as a separate sentence?

> Waits until no error markers are found for the specified projects.  Use all projects in the workspace if {@code projects} is empty.  _Returns {@code true} **when no error markers are found and** {@code false} after time-out of 300 seconds.